drv/lpc55-update-server/src/main.rs

```diff
@@ -273,14 +294,55 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
         Ok(info)
     }
 
+    fn versioned_rot_boot_info(
+        &mut self,
+        _: &RecvMessage,
+        version: u8,
+    ) -> Result<VersionedRotBootInfo, RequestError<UpdateError>> {
+        let boot_state =
+            bootstate().map_err(|_| UpdateError::MissingHandoffData)?;
+        let (
+            persistent_boot_preference,
+            pending_persistent_boot_preference,
+            transient_boot_preference,
+        ) = self.boot_preferences()?;
+
+        match version {
+            // There are deprecated versions
+            0 => Err(UpdateError::VersionNotSupported.into()),
+            1 => Ok(VersionedRotBootInfo::V1(RotBootInfo {
+                active: boot_state.active.into(),
+                persistent_boot_preference,
+                pending_persistent_boot_preference,
+                transient_boot_preference,
+                slot_a_sha3_256_digest: Some(boot_state.a.digest),
+                slot_b_sha3_256_digest: Some(boot_state.b.digest),
+            })),
+            _ => Ok(VersionedRotBootInfo::V2(RotBootInfoV2 {
```

Comment:
This is forward compatibility. If there is a next version that this implementation doesn't know about it will return the highest that it does know about. That allows the control plane to continue the update procedure without another roundtrip.

Reply:
Ah, I see --- that makes sense. Maybe a comment explaining that would be nice?

Reply:
done